packages/typespec-ts/src/modular/helpers/typeHelpers.ts

```diff
@@ -1,155 +1,231 @@
 import { EnumValue, Type } from "../modularCodeModel.js";
 
+/**
+ * Represents metadata for a given Type to generate the TypeScript equivalent.
+ */
 export interface TypeMetadata {
   name: string;
   originModule?: string;
   isRelative?: boolean;
+  nullable?: boolean;
   modifier?: "Array";
 }
 
-function getNullableType(name: string, type: Type): string {
-  if (type.nullable) {
-    return `(${name} | null)`;
-  }
-
-  return name;
-}
-
+/**
+ * Returns a string representation for an anonymous enum using its values.
+ */
 function getAnonymousEnumName(values: EnumValue[]): string {
   return values
     .map((v) => (typeof v.value === "string" ? `"${v.value}"` : `${v.value}`))
     .join(" | ");
 }
 
+// Mapping of simple types to their TypeScript equivalents.
+const simpleTypeMap: Record<string, TypeMetadata> = {
+  Key: {
+    name: "KeyCredential",
+    originModule: "@azure/core-auth",
+    isRelative: false
+  },
+  OAuth2: {
+    name: "TokenCredential",
+    originModule: "@azure/core-auth",
+    isRelative: false
+  },
+  boolean: { name: "boolean" },
+  datetime: { name: "Date" },
+  float: { name: "number" },
+  integer: { name: "number" },
+  "byte-array": { name: "Uint8Array" },
+  string: { name: "string" },
+  any: { name: "Record<string, any>" },
+  unknown: { name: "unknown" }
+};
+
+function getTypeName(type: { name?: string; nullable?: boolean }): string {
+  if (!type.name) {
+    throw new Error("Unable to process type without name");
+  }
+  let name = type.name;
+
+  if (type.nullable) {
+    name = `(${name} | null)`;
```

Comment:
If the `getTypeName` method only handles nullable logic why not change them like handleNullableName?